python/triton/compiler/compiler.py

```diff
@@ -91,12 +91,17 @@ def parse_options(self):
 
 class IRSource:
 
-    def __init__(self, path, context):
+    def __init__(self, path, context, target=None):
         self.path = path
         path = Path(path)
         self.ext = path.suffix[1:]
         self.src = path.read_text()
         ir.load_dialects(context)
+        if target is None:
+            target = driver.active.get_current_target()
+        assert isinstance(target, GPUTarget), "target must be of GPUTarget type"
```

Comment:
Why the target must be a GPU target?

Reply:
I took this check from `compile` function, just in case. However, looking at the code, it seems I can make `target` (or passing `backend` directly as suggested by @peterbell10 ) non-optional, then there will be no need to have `assert` here. Does it make sense?